docs/source/plotting/MeshPlotHelp.rst

```diff
@@ -76,7 +81,294 @@ Basic example of plotting a general Poly3DCollection `Polygon <https://matplotli
     plt.show()
 
 
-For more advice: :ref:`02_scripting_plots`
+CSGObject
+#########
+
+For help defining CSG Shapes and Rotations, see :ref:`HowToDefineGeometricShape`.
+Plot a CSGObject defined in an XML string:
+
+.. plot::
+   :include-source:
+
+   # import mantid algorithms, numpy and matplotlib
+   from mantid.simpleapi import *
+   import matplotlib.pyplot as plt
+   import numpy as np
+   from mpl_toolkits.mplot3d.art3d import Poly3DCollection
+
+   ws = CreateSampleWorkspace()
+
+   merge_xml = ' \
+   <cylinder id="stick"> \
+   <centre-of-bottom-base x="-0.5" y="0.0" z="0.0" /> \
+   <axis x="1.0" y="0.0" z="0.0" />  \
+   <radius val="0.05" /> \
+   <height val="1.0" /> \
+   </cylinder> \
+   \
+   <sphere id="some-sphere"> \
+   <centre x="0.7"  y="0.0" z="0.0" /> \
+   <radius val="0.2" /> \
+   </sphere> \
+   \
+   <rotate-all x="90" y="-45" z="0" /> \
+   <algebra val="some-sphere (: stick)" /> \
+   '
+
+   SetSample(ws, Geometry={'Shape': 'CSG', 'Value': merge_xml})
+
+   sample = ws.sample()
+   shape = sample.getShape()
+   mesh = shape.getMesh()
+
+   mesh_polygon = Poly3DCollection(mesh, edgecolors = 'blue', linewidths=0.1)
+   mesh_polygon.set_facecolor((1,0,0,0.5))
+
+   fig, axes = plt.subplots(subplot_kw={'projection':'mantid3d'})
+   axes.add_collection3d(mesh_polygon)
+
+   # Auto scale to the mesh size
+   axes_lims = mesh.flatten()
+   axes.auto_scale_xyz(axes_lims, axes_lims, axes_lims)
+   axes.view_init(elev=10, azim=-150)
+
+   axes.set_title('Sample Shape: Microphone')
+   axes.set_xlabel('X / m')
+   axes.set_ylabel('Y / m')
+   axes.set_zlabel('Z / m')
+
+   plt.show()
+
+Containers and Components
+#########################
+
+For help defining Containers and Components, see :ref:`SampleEnvironment`.
+Note Component index 0 is usually the Container.
+
+.. code-block:: python
+
+   ws = CreateSampleWorkspace()
+   LoadInstrument(Workspace=ws,RewriteSpectraMap=True,InstrumentName="Pearl")
+   SetSample(ws, Environment={'Name': 'Pearl'})
+
+   sample = ws.sample()
+   environment = sample.getEnvironment()
+
+   '''getMesh() to plot the Sample Shape'''
+   mesh = sample.getShape().getMesh()
+
+   '''getMesh() to plot the Container Shape'''
+   container_mesh = environment.getContainer().getShape().getMesh()
+
+   '''getMesh() to plot any Component Shape'''
+   # Component index 0 is the Container:
+   # container_mesh = environment.getComponent(0).getShape().getMesh()
+   component_mesh = environment.getComponent(2).getMesh()
+
```

Comment:
This script doesn't plot anything at the moment. I'm guessing this is deliberate because it avoids repeating the plotting lines already outlined twice higher up. Perhaps a single line comment in the script saying "...plot as previously described" or similar would make this totally clear